pkg/controller/statusmanager/kube.go

```diff
@@ -0,0 +1,124 @@
+package statusmanager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift/cluster-network-operator/pkg/apply"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Some silly types so that we can actually do testing
+
+type DaemonSetLister interface {
+	List(selector labels.Selector) (ret []*appsv1.DaemonSet, err error)
+}
+
+type DeploymentLister interface {
+	List(selector labels.Selector) (ret []*appsv1.Deployment, err error)
+}
+
+type StatefulSetLister interface {
+	List(selector labels.Selector) (ret []*appsv1.StatefulSet, err error)
+}
+
+type patchAnnotations struct {
+	Metadata md `json:"metadata"`
+}
+type md struct {
+	Annotations map[string]interface{} `json:"annotations"`
+}
+
+func makeAnnotationPatch(key string, value *string) ([]byte, error) {
+	patch := &patchAnnotations{
```

Comment:
FWIW you could do

    patch := &metav1.PartialObjectMetadata{

Reply:
oh nice!